src/main/java/it/gov/pagopa/pu/workflow/wf/classification/iuf/classification/IufClassificationWFImpl.java

```diff
@@ -0,0 +1,136 @@
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static it.gov.pagopa.pu.workflow.wf.ingestionflow.paymentsreporting.wfingestion.PaymentsReportingIngestionWFImpl.TASK_QUEUE;
+
+/**
+ * Implementation of the IufClassificationWF interface
+ */
+@Slf4j
+@WorkflowImpl(taskQueues = TASK_QUEUE)
+public class IufClassificationWFImpl implements IufClassificationWF, ApplicationContextAware {
+  public static final String TASK_QUEUE = "IufClassificationWF";
+
+  private IufClassificationNotifyTreasurySignalDTO signalDTO;
+
+  private ClearClassifyIufActivity clearClassifyIufActivity;
+  private IufClassificationActivity iufClassificationActivity;
+  private TransferClassificationActivity transferClassificationActivity;
+
+  private StartTransferClassificationActivity startTransferClassificationActivity;
+
+  private IufClassificationActivityResult result;
+
+  /**
+   * Temporal workflow will not allow to use injection in order to avoid
+   * <a href="https://docs.temporal.io/workflows#non-deterministic-change">non-deterministic changes</a> due to dynamic reconfiguration.<BR />
+   * Anyway it allows to override ActivityOptions, but actually it's not supporting the override based on the particular workflow.<BR />
+   * In {@link it.gov.pagopa.pu.workflow.config.TemporalWFImplementationCustomizer} we are already setting defaults to all workflows.<BR />
+   * Use this as an example to override based on the particular workflow.
+   */
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    IufClassificationWfConfig wfConfig = applicationContext.getBean(IufClassificationWfConfig.class);
+
+    clearClassifyIufActivity = wfConfig.buildClearClassifyIufActivityStub();
+    iufClassificationActivity = wfConfig.buildIufClassificationActivityStub();
+
+    startTransferClassificationActivity = wfConfig.buildTransferClassificationStarterHelperActivityStub();
+
+  }
+
+  @Override
+  public void classify() {
+
+    if (result != null && result.isSuccess()) {
+
+      Set<Transfer2ClassifyDTO> uniqueTransfers = new HashSet<>(result.getTransfers2classify());
+
+      // signal transfer classification for each transfer
+      uniqueTransfers.forEach(transfer2ClassifyDTO -> {
+
+        String iuv = transfer2ClassifyDTO.getIuv();
+        String iur = transfer2ClassifyDTO.getIur();
+        int transferIndex = transfer2ClassifyDTO.getTransferIndex();
+
+        startTransferClassificationActivity.signalTransferClassificationWithStart(result.getOrganizationId(), iuv, iur, transferIndex);
+      });
+
+    } else {
+      log.warn("Result is null or classification was not successful for organizationId {}",
+        result != null ? result.getOrganizationId() : "N/A");
+    }
```

Comment:
let's ignore not successful
```suggestion
    toNotify.stream()
       .flatMap(r -> r. getTransfers2classify().stream())
       .distinct()
       ..forEach(transfer2ClassifyDTO -> {

        String iuv = transfer2ClassifyDTO.getIuv();
        String iur = transfer2ClassifyDTO.getIur();
        int transferIndex = transfer2ClassifyDTO.getTransferIndex();

        startTransferClassificationActivity.signalTransferClassificationWithStart(toNotify.get(0).getOrganizationId(), iuv, iur, transferIndex);
      });
```

Reply:
OK